### PR TITLE
Migrating the measurement and publish workflows to GitHub Actions

### PR DESCRIPTION
**Pull request by smiasojed**
## Summary
Related #1454
- [n] y/n | Does it introduce breaking changes?
- [n] y/n | Is it dependant on the specific version of `cargo-contract` or `pallet-contracts`?
Migrating the measurement and publish workflows to GitHub Actions

## Description
Depends on https://github.com/paritytech/ink/pull/1977
Adds the measurement and publish-docs workflows

## Checklist before requesting a review
- [y] My code follows the style guidelines of this project
- [n] I have added an entry to `CHANGELOG.md`
- [y] I have commented my code, particularly in hard-to-understand areas
- [n] I have added tests that prove my fix is effective or that my feature works
- [y] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
## 🦑 📈 ink! Example Contracts ‒ Changes Report 📉 🦑
⚠️ **The ink! `master` is ahead of your branch, this might skew the comparison data below.** ⚠️
These are the results when building the `integration-tests/*` contracts from this branch with `cargo-contract` and comparing them to ink! `master`:

| Contract                                                | Upstream Size (kB) | PR Size (kB) | Diff (kB) | Diff (%)  | Change                     |
| ------------------------------------------------------- | ------------------ | ------------ | --------- | --------- | -------------------------- |
| basic-contract-caller                                   | 2.967              | 2.967        | 0         | 0         | :heavy_minus_sign:         |
| basic-contract-caller/other-contract                    | 1.337              | 1.337        | 0         | 0         | :heavy_minus_sign:         |
| call-builder-return-value                               | 8.735              | 8.735        | 0         | 0         | :heavy_minus_sign:         |
| call-runtime                                            | 1.769              | 1.769        | 0         | 0         | :heavy_minus_sign:         |
| conditional-compilation                                 | 1.209              | 1.209        | 0         | 0         | :heavy_minus_sign:         |
| contract-storage                                        | 7.171              | 7.171        | 0         | 0         | :heavy_minus_sign:         |
| contract-terminate                                      | 1.092              | 1.092        | 0         | 0         | :heavy_minus_sign:         |
| contract-transfer                                       | 1.444              | 1.444        | 0         | 0         | :heavy_minus_sign:         |
| custom-allocator                                        | 7.428              | 7.428        | 0         | 0         | :heavy_minus_sign:         |
| dns                                                     | 7.249              | 7.249        | 0         | 0         | :heavy_minus_sign:         |
| e2e-call-runtime                                        | 1.058              | 1.058        | 0         | 0         | :heavy_minus_sign:         |
| e2e-runtime-only-backend                                | 1.635              | 1.635        | 0         | 0         | :heavy_minus_sign:         |
| erc1155                                                 | 13.962             | 13.968       | 0.006     | 0.0429738 | :chart_with_upwards_trend: |
| erc20                                                   | 6.687              | 6.687        | 0         | 0         | :heavy_minus_sign:         |
| erc721                                                  | 9.64               | 9.64         | 0         | 0         | :heavy_minus_sign:         |
| events                                                  | 4.763              | 4.763        | 0         | 0         | :heavy_minus_sign:         |
| flipper                                                 | 1.393              | 1.393        | 0         | 0         | :heavy_minus_sign:         |
| incrementer                                             | 1.221              | 1.221        | 0         | 0         | :heavy_minus_sign:         |
| lang-err-integration-tests/call-builder-delegate        | 2.317              | 2.317        | 0         | 0         | :heavy_minus_sign:         |
| lang-err-integration-tests/call-builder                 | 4.847              | 4.847        | 0         | 0         | :heavy_minus_sign:         |
| lang-err-integration-tests/constructors-return-value    | 1.773              | 1.773        | 0         | 0         | :heavy_minus_sign:         |
| lang-err-integration-tests/contract-ref                 | 4.328              | 4.328        | 0         | 0         | :heavy_minus_sign:         |
| lang-err-integration-tests/integration-flipper          | 1.571              | 1.571        | 0         | 0         | :heavy_minus_sign:         |
| mapping-integration-tests                               | 7.685              | 7.685        | 0         | 0         | :heavy_minus_sign:         |
| mother                                                  | 9.508              | 9.508        | 0         | 0         | :heavy_minus_sign:         |
| multi-contract-caller                                   | 5.924              | 5.924        | 0         | 0         | :heavy_minus_sign:         |
| multi-contract-caller/accumulator                       | 1.095              | 1.095        | 0         | 0         | :heavy_minus_sign:         |
| multi-contract-caller/adder                             | 1.669              | 1.669        | 0         | 0         | :heavy_minus_sign:         |
| multi-contract-caller/subber                            | 1.689              | 1.689        | 0         | 0         | :heavy_minus_sign:         |
| multisig                                                | 21.471             | 21.471       | 0         | 0         | :heavy_minus_sign:         |
| payment-channel                                         | 5.501              | 5.501        | 0         | 0         | :heavy_minus_sign:         |
| sr25519-verification                                    | 0.865              | 0.865        | 0         | 0         | :heavy_minus_sign:         |
| static-buffer                                           | 1.405              | 1.405        | 0         | 0         | :heavy_minus_sign:         |
| trait-dyn-cross-contract-calls                          | 2.466              | 2.466        | 0         | 0         | :heavy_minus_sign:         |
| trait-dyn-cross-contract-calls/contracts/incrementer    | 1.305              | 1.305        | 0         | 0         | :heavy_minus_sign:         |
| trait-erc20                                             | 7.063              | 7.063        | 0         | 0         | :heavy_minus_sign:         |
| trait-flipper                                           | 1.209              | 1.209        | 0         | 0         | :heavy_minus_sign:         |
| trait-incrementer                                       | 1.37               | 1.37         | 0         | 0         | :heavy_minus_sign:         |
| upgradeable-contracts/delegator                         | 2.908              | 2.908        | 0         | 0         | :heavy_minus_sign:         |
| upgradeable-contracts/delegator/delegatee               | 1.369              | 1.369        | 0         | 0         | :heavy_minus_sign:         |
| upgradeable-contracts/set-code-hash                     | 1.464              | 1.464        | 0         | 0         | :heavy_minus_sign:         |
| upgradeable-contracts/set-code-hash/updated-incrementer | 1.443              | 1.443        | 0         | 0         | :heavy_minus_sign:         |
| wildcard-selector                                       | 2.622              | 2.622        | 0         | 0         | :heavy_minus_sign:         |

[Link to the run](https://gitlab.parity.io/parity/mirrors/ink/-/pipelines/408965) | Last update: Fri Nov  3 12:46:29 CET 2023

File: .gitlab-ci.yml
Comment (ascjones) [inline .gitlab-ci.yml:1]: For this stage I reckon we should just delete it from the gitlab runner already.

Then we should delete the contract sizes comment from this PR and we can prove it is working for sure.

Comment (smiasojed) [reply .gitlab-ci.yml:1]: To test it correctly we need to merge https://github.com/paritytech/ink/pull/1977 first
We do not need to delete comment, just will be added another one. Now comments are submitted from 'github-actions[bot]' account

**Comment by smiasojed** [comment]
> ## [Codecov](https://app.codecov.io/gh/paritytech/ink/pull/1971?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech) Report
> > Merging [#1971](https://app.codecov.io/gh/paritytech/ink/pull/1971?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech) ([2025907](https://github.com/paritytech/ink/commit/2025907309a208b8c0feb6b5a620e472e1427c92)) into [master](https://app.codecov.io/gh/paritytech/ink/commit/d125d7eb451c9a7a6276dcc163ae2e75b5e0766f?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech) ([d125d7e](https://github.com/paritytech/ink/commit/d125d7eb451c9a7a6276dcc163ae2e75b5e0766f)) will **increase** coverage by `0.02%`.
> > The diff coverage is `n/a`.
> 
> ```diff
> @@            Coverage Diff             @@
> ##           master    #1971      +/-   ##
> ==========================================
> + Coverage   53.31%   53.34%   +0.02%     
> ==========================================
>   Files         219      219              
>   Lines        6835     6835              
> ==========================================
> + Hits         3644     3646       +2     
> + Misses       3191     3189       -2     
> ```
> 
> [see 2 files with indirect coverage changes](https://app.codecov.io/gh/paritytech/ink/pull/1971/indirect-changes?src=pr&el=tree-more&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech)
> 
> 📣 Codecov offers a browser extension for seamless coverage viewing on GitHub. Try it in [Chrome](https://chrome.google.com/webstore/detail/codecov/gedikamndpbemklijjkncpnolildpbgo) or [Firefox](https://addons.mozilla.org/en-US/firefox/addon/codecov/) today!

@s

> ## [Codecov](https://app.codecov.io/gh/paritytech/ink/pull/1971?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech) Report
> > Merging [#1971](https://app.codecov.io/gh/paritytech/ink/pull/1971?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech) ([2025907](https://github.com/paritytech/ink/commit/2025907309a208b8c0feb6b5a620e472e1427c92)) into [master](https://app.codecov.io/gh/paritytech/ink/commit/d125d7eb451c9a7a6276dcc163ae2e75b5e0766f?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech) ([d125d7e](https://github.com/paritytech/ink/commit/d125d7eb451c9a7a6276dcc163ae2e75b5e0766f)) will **increase** coverage by `0.02%`.
> > The diff coverage is `n/a`.
> 
> ```diff
> @@            Coverage Diff             @@
> ##           master    #1971      +/-   ##
> ==========================================
> + Coverage   53.31%   53.34%   +0.02%     
> ==========================================
>   Files         219      219              
>   Lines        6835     6835              
> ==========================================
> + Hits         3644     3646       +2     
> + Misses       3191     3189       -2     
> ```
> 
> [see 2 files with indirect coverage changes](https://app.codecov.io/gh/paritytech/ink/pull/1971/indirect-changes?src=pr&el=tree-more&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=paritytech)
> 
> 📣 Codecov offers a browser extension for seamless coverage viewing on GitHub. Try it in [Chrome](https://chrome.google.com/webstore/detail/codecov/gedikamndpbemklijjkncpnolildpbgo) or [Firefox](https://addons.mozilla.org/en-US/firefox/addon/codecov/) today!

@sergejparity could you please take a look into codecov job if it looks good for you?
I see that we have:
https://app.codecov.io/gh/paritytech-ci/ink/pulls
https://app.codecov.io/gh/paritytech/ink/pulls
could you please clarify what is the difference between them?

**Review by ascjones — APPROVED** [review]
LGTM

**Review by sergejparity — APPROVED** [review]
LGTM

**Pull request merged**